IRescue/UserLocalisation/Sensors/Marker/MarkerSensor.cs

```diff
@@ -43,25 +53,44 @@ public MarkerSensor(float standardDeviation, string path)
         {
             this.standardDeviation = standardDeviation;
             this.markerLocations = new MarkerLocations(path);
+            this.orientations = new Measurement<Vector3>[this.bufferLength];
+            this.positions = new Measurement<Vector3>[this.bufferLength];
+            this.Measurements = 0;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MarkerSensor"/> class with a given buffer size.
+        /// </summary>
+        /// <param name="standardDeviation">the standard deviation</param>
+        /// <param name="path">url to the xml file</param>
+        /// <param name="bufferLength">Length of the buffer</param>
+        public MarkerSensor(float standardDeviation, string path, int bufferLength) : this(standardDeviation, path)
+        {
+            this.bufferLength = bufferLength;
         }
 
         /// <summary>
+        /// Gets the amount of measurements.
+        /// </summary>
+        public int Measurements { get; private set; }
+
+        /// <summary>
         ///   Update the locations derived from Markers.
         /// </summary>
         /// <param name="visibleMarkerIds">Dictionary of the ids and transforms ((x,y,z), (pitch, yaw, rotation) in degrees) of the visible Markers.</param>
         public void UpdateLocations(Dictionary<int, Pose> visibleMarkerIds)
         {
             long timeStamp = StopwatchSingleton.Time;
-            this.positions = new List<Measurement<Vector3>>(visibleMarkerIds.Count);
-            this.orientations = new List<Measurement<Vector3>>(visibleMarkerIds.Count);
             foreach (KeyValuePair<int, Pose> pair in visibleMarkerIds)
             {
                 try
                 {
                     Pose currentMarkerPose = this.markerLocations.GetMarker(pair.Key);
                     Pose location = AbRelPositioning.GetLocation(currentMarkerPose, pair.Value);
-                    this.positions.Add(new Measurement<Vector3>(location.Position, this.standardDeviation, timeStamp));
-                    this.orientations.Add(new Measurement<Vector3>(location.Orientation, this.standardDeviation, timeStamp));
+                    this.positions[this.pointer] = new Measurement<Vector3>(location.Position, this.standardDeviation, timeStamp);
+                    this.orientations[this.pointer] = new Measurement<Vector3>(location.Orientation, this.standardDeviation, timeStamp);
+                    this.pointer = this.pointer >= this.bufferLength - 1 ? 0 : this.pointer + 1;
+                    this.Measurements++;
```

Comment:
Depends on what the intention of the value is. To me its current name and documentation suggests the number of measurements currently stored in the sensor.